src/hotspot/share/gc/g1/g1FullCollector.hpp

```diff
@@ -76,12 +76,15 @@ class G1FullCollector : StackObj {
   G1FullGCScope             _scope;
   uint                      _num_workers;
   bool                      _has_compaction_targets;
+  bool                      _has_humongous;
   G1FullGCMarker**          _markers;
   G1FullGCCompactionPoint** _compaction_points;
   OopQueueSet               _oop_queue_set;
   ObjArrayTaskQueueSet      _array_queue_set;
   PreservedMarksSet         _preserved_marks_set;
   G1FullGCCompactionPoint   _serial_compaction_point;
+  G1FullGCCompactionPoint   _humongous_compaction_point;
+  GrowableArray<HeapRegion*>* _humongous_compaction_regions;
```

Comment:
This could be a `GrowableArrayCHeap<HeapRegion*, mtGC> _humongous_compaction_region` to avoid the necessity for explicit new/delete. Afaics there is no reassignment of that variable ever.